improve throttle

anasmatic opened the pull request:
it is much smoother now, hopefully will remove the throttle bumping effect the grader talked about in submit 1 result video.
the throttle diagram in submit one was like this : 
![throttle_1 0_0 0_0 5](https://user-images.githubusercontent.com/3892671/38771711-6be89f8c-405a-11e8-8013-900fc3703bbd.png)

now it is smoother with little overshooting when we hit the max speed, but brakes on traffic lights is firm without jerking .
![throttle_0 2_0 0005_0 05](https://user-images.githubusercontent.com/3892671/38771724-b2857ca8-405a-11e8-8ed8-f034731bd8cc.png)

anasmatic committed:
improve throttle

is much smoother.

anasmatic committed:
improve throttle

is much smoother.

clockworkbits commented:
I tested it in both highway and test lot scenarios. Highway is fine, although I haven't waited till the end. Test lot is not - the car fails to stop at the last waypoint and in result drives in a loop.

agotterba commented:
I ran it to the end of the simulator track, and it did stop successfully.  I was trying to test traffic light performance, but saw that there was too much lag, and detections were iffy with the parking lot classifier.  Once I switched to the simulator classifier, it stops at the stop-lines correctly, though the lag is even worse.

anasmatic committed:
using lowpassfilter + add debug lines

debug lines will collect data from throttle and steer , save in array,
so I can analyse them later

anasmatic committed:
using lowpassfilter + add debug lines

debug lines will collect data from throttle and steer , save in array,
so I can analyse them later

anasmatic commented:
after using lowpass filter , throttle less smother but is ok , 
it can stop at the end waypoints , and traffic-lights with no issues 
![throttle_1 0_0 0_0 5_f100](https://user-images.githubusercontent.com/3892671/38889852-7326bee6-42b2-11e8-9d0a-7fce0b129cde.png)

agotterba reviewed (approved):
I simulated it on the parking lot track, and see the car stop at the end of the waypoints now (and that it didn't stop when using the previous commit)

anasmatic commented:
thank you for the approval : 

1 - if `is_dbw_enabled = False` 
then jerk at start will be around **500±** , 
Throttle graph shows that we start with highest value .
![init_true_isdbw](https://user-images.githubusercontent.com/3892671/38932156-f5a162b4-4347-11e8-9397-2d84cfc374d0.png)

2 - if `is_dbw_enabled = True` and wait for one minute before switch manual,
 then jerk at start will be around **200±** , 
Throttle graph shows that we start with gradual throttle to reach highest value ,
the more we wait the less time it takes to reach highest throttle value.
End throttle will not be smooth , 
![ture_isdbw](https://user-images.githubusercontent.com/3892671/38932419-b3e06f2c-4348-11e8-97c1-7171a5987237.png)


3 - if `is_dbw_enabled = True` and didn't wait before switch 
then jerk at start will be very low  , 
Throttle graph shows that we start with good gradual throttle to reach highest value ,
![false_isdbw](https://user-images.githubusercontent.com/3892671/38932468-dbef0ca8-4348-11e8-8439-b719f9d8bc72.png)

the best is _situation 1_ ( `is_dbw_enabled = True` )
even-though _situation 3_ will have the best start, but waiting before switching to auto will eliminate all its cons.

anasmatic committed:
disable dbw on init + lowpass more smooth

set is_dbw_enabled = False as initialization value

+ set lowpass filter to 500 gives more smooth throttle

anasmatic committed:
rebase update

anasmatic commented:
Actually the smooth part at the start is fake _ -when `is_dbw_enable=True`- _ , it is calculated before switching from manual to auto  

still has the same question, any ideas how to smooth throttle start so we won't "spun the wheels"

clockworkbits commented on ros/src/twist_controller/dbw_node.py:
```diff
@@ -33,7 +33,7 @@
 '''
 
-DEBUG_MODE            = False
+DEBUG_MODE            = True
```
Is this intended?

clockworkbits commented:
In the walkthrough video there is a lowpass filter with quite low tau applied to the current velocity. Maybe we should try to do similar thing? Not sure if this will help much. But I'm pretty sure we should start with `is_dbw_enabled = False`
